src/precice/impl/Participant.cpp

```diff
@@ -37,8 +38,14 @@ Participant::~Participant()
     delete context;
   }
   _usedMeshContexts.clear();
-  _readDataContexts.deleteElements();
-  _writeDataContexts.deleteElements();
+  for (ReadDataContext *elem : _readDataContexts) {
+    PRECICE_ASSERT(elem != NULL);
+    delete (elem);
+  }
+  for (DataContext *elem : _writeDataContexts) {
+    PRECICE_ASSERT(elem != NULL);
+    delete (elem);
+  }
```

Comment:
This code cries for smart pointers.

Reply:
`PointerVector.hpp` is already crying for quite some time :smirk: 

But let's try to clean this up and do it better here. I agree. I just copy-pasted this part without a lot of thinking.

Reply:
Fixed in ce0eb84.